### PR TITLE
Update the basic Python scaffold

### PR DESCRIPTION
**Pull request by BrunoGrandePhD**
The goal of this PR is to adapt the package skeleton provided by PyScaffold for `dcqc`. Some of these changes were copied from `sagetasks`, which was also packaged using PyScaffold. Here are the main changes:

- Use CodeCov instead of Coveralls for tracking code coverage
  - I was having issues when I tried using Coveralls with `sagetasks`
- Add `mypy` (static type checker) as a pre-commit hook
- Create a `Pipfile` that installs the package and its dependencies
- Update the README and CONTRIBUTING docs

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -147,27 +96,24 @@ conda activate dcqc
 
 3. Clone this copy to your local disk:
 
-   ```
-   git clone git@github.com:YourLogin/dcqc.git
+   ```console
+   git clone git@github.com:sage-bionetworks-workflows/dcqc.git
```
Comment (thomasyu888) [inline CONTRIBUTING.md:100]: Fix git clone path.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -357,15 +287,9 @@ on [PyPI], the following steps can be used to release a new version for
...
 [virtualenv]: https://virtualenv.pypa.io/en/stable/
-
-
-```{todo} Please review and change the following definitions:
-```
-
-[repository]: https://github.com/<USERNAME>/dcqc
-[issue tracker]: https://github.com/<USERNAME>/dcqc/issues
+[repository]: https://github.com/sage-bionetworks-workflows/dcqc
+[issue tracker]: https://github.com/sage-bionetworks-workflows/dcqc/issues
```
Comment (thomasyu888) [inline CONTRIBUTING.md:295]: Fix these links

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -67,29 +40,22 @@ by adding missing information and correcting mistakes.
...
-      would like to propose changes and click in the little pencil icon at the
-      top, to open [GitHub's code editor]. Once you finish editing the file,
-      please write a message in the form at the bottom of the page describing
-      which changes have you made and what are the motivations behind them and
-      submit your proposal.
-   :::
-```
+:::{tip}
+   Please notice that the [GitHub web interface] provides a quick way of
+   propose changes in `dcqc`'s files. While this mechanism can
```
Comment (thomasyu888) [inline CONTRIBUTING.md:47]: `dcqc` or `py-dcqc`?

**Review by thomasyu888 — APPROVED** [review]
LGTM - minor comments, but going to approve first.

File: CONTRIBUTING.md (hunk 3 above)
Comment (BrunoGrandePhD) [reply CONTRIBUTING.md:47]: When referring to the package, I'll just use `dcqc`. The repo has `py-` to differentiate it from the eventualy `nf-dcqc` repo.

**Pull request merged**